worker.js

```diff
@@ -16692,7 +16692,7 @@ class HTMLDocument extends Document$1 {
   get head() {
     const {documentElement} = this;
     let {firstElementChild} = documentElement;
-    if (!firstElementChild) {
+    if (!firstElementChild || firstElementChild.tagName !== 'HEAD') {
```

Comment:
> [...] plus it's not considering that the current logic works already for document.body so it's kinda patching it in the wrong place.
> 
> The body getter is likely the place to look for a change, not the parse-from-string one, as that file has zero responsibility on how that document or its getters should react .. and bear in mind document.head is another one to eventually fix in the original issue.

Is this more what you mean?

---

Previously, the code would always assume the that first child was a `<head>` element but in the case of `<!DOCTYPE html><html><body><div>foo</div></body></html>`, it's a `<body>`. This means that when the `body` getter calls the `head` getter, it returned the existing `<body>` and created another `<body>` for itself after the supposed `<head>`.

The new code makes sure that the first child is a `<head>` element otherwise creates one.

This current assumptions about node order here has the downside that if someone defines stuff out of order, it will have 2 `<head>` when you try to access `document.head`

```js
// notice <head> after <body>
const document = new DOMParser().parseFromString(`<!DOCTYPE html><html><body></body><head><title>"hello"</title></head></html>`, "text/html").head.outerHTML
// '<head></head>'
```

We could fix this up better in `parse-from-string` if we enforce `<head>` as the first child and `<body>` after when documents are initially parsed and created. But I'd go back to the `document.body` creation pattern like I had previously to keep things simple.